modules/social_features/social_profile/modules/social_profile_privacy/src/Service/SocialProfilePrivacyBatchHelper.php

```diff
@@ -0,0 +1,258 @@
+    $private_fields_list = social_profile_privacy_private_fields_list($uid);
+
+    $account_name_fields = [
+      'field_profile_first_name',
+      'field_profile_last_name',
+    ];
+
+    foreach ($account_name_fields as $key => $account_name_field) {
+      if (in_array($account_name_field, $private_fields_list)) {
+        unset($account_name_fields[$key]);
+      }
+      else {
+        $account_name_fields[$key] = $profile->get($account_name_field)->getString();
+      }
+    }
+
+    if (!empty($account_name_fields) && !empty($full_name = implode(" ", $account_name_fields))) {
+      $account_name = $full_name;
+    }
+
+    if ($this->moduleHandler->moduleExists('social_profile_fields') && !in_array('field_profile_nick_name', $private_fields_list) && !empty($profile->get('field_profile_nick_name')->getString())) {
+      $account_name = $profile->get('field_profile_nick_name')->getString();
+    }
+
+    return $account_name;
+  }
+
+  /**
+   * Update profile names in a batch.
+   */
+  public static function bulkUpdateProfileNames() {
+    // Define batch process to update profile names.
+    $batch_builder = (new BatchBuilder())
+      ->setTitle(t('Updating profile names...'))
+      ->setFinishCallback([
+        SocialProfilePrivacyBatchHelper::class,
+        'finishProcess',
+      ])
+      ->addOperation([SocialProfilePrivacyBatchHelper::class, 'initOperation'], [
+        ['limit' => 50],
+      ])
+      ->addOperation([SocialProfilePrivacyBatchHelper::class, 'updateProcess'], [
+        ['limit' => 50],
+      ]);
+
+    batch_set($batch_builder->toArray());
+  }
+
+  /**
+   * Init operation task by retrieving all content to be updated.
+   *
+   * @param array $args
+   *   Arguments.
+   * @param array $context
+   *   An array that may or may not contain placeholder variables.
+   */
+  public static function initOperation(array $args, array &$context) {
+    // Init variables.
+    $limit = (int) $args['limit'];
+    $offset = (int) $context['sandbox'] ?? 0;
+
+    $userStorage = \Drupal::entityTypeManager()->getStorage('user');
+
+    $uids = $userStorage->getQuery()
+      ->condition('status', '1')
+      ->execute();
+    /** @var array $results */
+    $results = $userStorage->loadMultiple($uids);
+
+    // Define total on first call.
+    if (!isset($context['sandbox']['total'])) {
+      $context['sandbox']['total'] = count($results);
+    }
+
+    // Setup results based on retrieved objects.
+    $context['results'] = array_reduce($results,
+      function ($carry, $object) {
+        // Map object results extracted from previous query.
+        $carry[$object->id()] = $object;
+        return $carry;
+      }, $context['results']
+    );
+
+    // Redefine offset value.
+    $context['sandbox']['offset'] = $offset + $limit;
+
+    // Set current step as unfinished until offset is greater than total.
+    $context['finished'] = 0;
+    if ($context['sandbox']['offset'] >= $context['sandbox']['total']) {
```

Comment:
What happens if the total profiles to be processed is 25 only?

Reply:
@Ressinel The limit is 50, what will happen in that this situation where the limit is more than the total number of results.

Reply:
I have changed a bit this batch code because found an issue for the case where items more than the limit. So check how it looks now.